### PR TITLE
34 Profile page, profile model and profile app

### PR DESCRIPTION
**Pull request by oshtohryn**
## TRELLO

* [Main Trello ticket](https://trello.com/b/BJDdC443/mytrip)

## Code reviewers

- [ ] @Ashiato
- [ ] @black-art
- [ ] @hrytskivr
- [ ] @KarnaushenkoVO
- [ ] @PanLeon
- [ ] @spycheast
- [ ] @yuriiZherebukh

### Second Level Review

- [ ] @byshovets
- [ ] @lhalam

## Summary of change
Created profile model and connect to CustomUser
Created views for profile
Created React profile component

## Testing approach

ToDo tests

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=90%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: myTrip/profile/migrations/0001_initial.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
```
Comment (lhalam) [inline myTrip/profile/migrations/0001_initial.py:1]: merge to one migration file

File: myTrip/profile/models.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    avatar = models.URLField(null=True)
+    age = models.PositiveIntegerField(null=True)
+    gender = models.TextField(null=True)
+    hobbies = models.TextField(null=True)
+    facebook = models.URLField(null=True)
+    instagram = models.URLField(null=True)
+    google = models.URLField(null=True)
+
+    @receiver(post_save, sender=CustomUser)
+    def create_user_profile(sender, instance, created, **kwargs):
```
Comment (lhalam) [inline myTrip/profile/models.py:37]: modifier

File: myTrip/profile/urls.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+from django.conf.urls import url, include
+from .views import ProfileView
+
+urlpatterns = [
+    url(r'^$', ProfileView.as_view(), name='profile'),
```
Comment (lhalam) [inline myTrip/profile/urls.py:5]: one urlpath

File: myTrip/profile/views.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+    def get(self, request, user_id=None):
+        """GET request handler. Can return profile
+         logged user or by given user_id."""
+        if user_id:
+            profile = Profile.get_by_id(user_id)
+            return JsonResponse(profile.to_dict(), status=200)
+        profile = Profile.get_by_id(request.user.id)
+        if not profile:
+            return HttpResponse(status=204)
```
Comment (lhalam) [inline myTrip/profile/views.py]: 404

File: myTrip/profile/views.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+class ProfileView(View):
+    """Class that handle HTTP requests."""
+
+    def get(self, request, user_id=None):
+        """GET request handler. Can return profile
+         logged user or by given user_id."""
+        if user_id:
+            profile = Profile.get_by_id(user_id)
+            return JsonResponse(profile.to_dict(), status=200)
+        profile = Profile.get_by_id(request.user.id)
```
Comment (lhalam) [inline myTrip/profile/views.py:22]: ` profile = request.user.profile `

File: myTrip/profile/views.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+            return HttpResponse(status=204)
+        return JsonResponse(profile.to_dict(), status=200)
+
+    def put(self, request):
+        """PUT request handler. Select logged
+         user profile and update it."""
+        profile = Profile.get_by_id(request.user.id)
+        if not profile:
+            return HttpResponse(status=403)
+        user = CustomUser.objects.get(id=request.user.id)
```
Comment (lhalam) [inline myTrip/profile/views.py:44]: ` user = request.user`

File: myTrip/static/src/profile/profile.js [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+import Paper from 'material-ui/Paper';
+import { Avatars } from './profileAvatar';
+import { Buttons } from './profileButtons';
+import { TextBlock } from './profileEdit';
+
+import './profile.less';
+
+const profileURL = '/api/v1/profile/';
+
```
Comment (lhalam) [inline myTrip/static/src/profile/profile.js:12]: delete one empty line

File: myTrip/static/src/profile/profile.js [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+
+export default class Profile extends React.Component {
+    constructor(props) {
+        super(props);
+        this.profileID = this.props.match.params.id;
```
Comment (lhalam) [inline myTrip/static/src/profile/profile.js]: `this.profileID = this.props.match.params.id || '';`

File: myTrip/static/src/profile/profile.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+        this.state = {
+            profile: null
+        };
+    }
+
+    getProfile = () => {
+        const profileID = this.profileID;
```
Comment (lhalam) [inline myTrip/static/src/profile/profile.js]:
```
   axios.get(profileURL + this.profileID)
        .then(response => {
            const profile = response.data;
            this.setState({profile: profile});
        });
```

File: myTrip/static/src/profile/profileByID.js [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+
+import './profile.less';
+
+const style = {
+    width: 500,
+    height: 500,
+}
+
+const profileURL = '/api/v1/profile/';
+const defaultAvatar = 'https://freeiconshop.com/wp-content/uploads/edd/person-flat.png'
```
Comment (lhalam) [inline myTrip/static/src/profile/profileByID.js]: delete

**Review by lhalam — COMMENTED** [review]
ToDo

File: myTrip/mytrip/urls.py [hunk 11]
```diff
@@ -21,8 +21,9 @@
     url(r'^api/v1/trip/', include('trip.urls')),
     url(r'^api/v1/auth/', include('registration.urls')),
     url(r'^api/v1/photo/', include('photo.urls')),
+    url(r'api/v1/help/', include('help.urls')),
```
Comment (black-art) [inline myTrip/mytrip/urls.py:24]: duplicate

**Review by black-art — APPROVED** [review]

**Pull request merged**